pkg/ingester-rf1/flush.go

```diff
@@ -144,19 +143,27 @@ func (i *Ingester) flushOp(l log.Logger, op *flushOp) error {
 // If the flush is successful, metrics for this flush are to be reported.
 // If the flush isn't successful, the operation for this userID is requeued allowing this and all other unflushed
 // segments to have another opportunity to be flushed.
-func (i *Ingester) flushSegment(ctx context.Context, ch *wal.SegmentWriter) error {
+func (i *Ingester) flushSegment(ctx context.Context, w *wal.SegmentWriter) error {
 	id := ulid.MustNew(ulid.Timestamp(time.Now()), rand.Reader)
-	r := ch.Reader()
 
 	start := time.Now()
 	defer func() {
-		runutil.CloseWithLogOnErr(util_log.Logger, r, "flushSegment")
 		i.metrics.flushDuration.Observe(time.Since(start).Seconds())
-		ch.Observe()
+		w.Observe()
 	}()
 
+	buf := i.flushBuffers.Get().(*bytes.Buffer)
```

Comment:
Nit: if we can avoid a pool by scoping this to the segment writer or the flush worker it's better

Reply:
We can do concurrent flushes (in the case where write volume is more than 8MB/sec) so I think we want a pool?

Reply:
Oh I misunderstood. Let me do that :)